apps/src/aichat/views/ChatWorkspace.tsx

```diff
@@ -50,12 +70,31 @@ const ChatWorkspace: React.FunctionComponent<ChatWorkspaceProps> = ({
     }
   }, [messagesString, isWaitingForChatResponse]);
 
-  const dispatch = useAppDispatch();
+  useEffect(() => {
+    let selectedStudentName = null;
+    if (viewAsUserId) {
+      const selectedStudent = Object.values(students).find(
+        student => student.id === viewAsUserId
+      );
+      if (selectedStudent) {
+        selectedStudentName = getShortName(selectedStudent.name);
+      }
+    }
+    setSelectedStudentName(selectedStudentName);
+  }, [viewAsUserId, students]);
```

Comment:
This is a double rendering anti-pattern.

Get rid of the `selectedStudentName` local variable and make it a `useMemo call instead.

```
const selectedStudentName = useMemo( () => {
    if (viewAsUserId) {
      const selectedStudent = Object.values(students).find(
        student => student.id === viewAsUserId
      );
      if (selectedStudent) {
        return getShortName(selectedStudent.name);
      }
    }
    return null
}, [viewAsUserId, students]);

Reply:
Thanks Jim! Will update.